src/EventGridExtension/OutputBinding/EventGridOutputAsyncCollector.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventGrid;
+using Microsoft.Azure.EventGrid.Models;
+
+namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
+{
+    public sealed class EventGridOutputAsyncCollector : IAsyncCollector<EventGridEvent>
+    {
+        private static EventGridClient _client;
+
+        private EventGridOutputAttribute _attribute;
+
+        public EventGridOutputAsyncCollector(EventGridOutputAttribute attr)
+        {
+            _attribute = attr;
+
+            if (_client == null)
+            {
+                _client = new EventGridClient(new TopicCredentials(_attribute.SasKey));
+            }
+        }
+
+        public Task AddAsync(EventGridEvent item, CancellationToken cancellationToken = default(CancellationToken)) => _client.PublishEventsAsync(_attribute.TopicHostname, new[] { item }, cancellationToken);
```

Comment:
You should use a `List<EventGridEvent>` for `AddAsync()` and call `PublishEventsAsync()` once in `FlushAsync()` to avoid unnecessary network traffics

Reply:
I debated this. Ultimately it was my thought that if a Function uses the output binding to send events to Event Grid, they want those events sent *immediately* upon `.AddAsync` to the collector, that's why I chose this route.
Do you think consumers could be confused by a whole batch of events coming thru at the end of a Function instead of as they were "sent" by the binding? That's where my mind went...

Reply:
I think the "flush" pattern is one customers have come accustomed to.  In other words if I *need* to have my function push events mid execution I'd be expected to "Flush", but by default all our other bindings don't flush until post-execution. Think it would be more confusing to have this one behave differently.